src/components/Card.tsx

```diff
@@ -0,0 +1,54 @@
+import '../styles/Card.scss';
+import { Link } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+
+export interface CardProps {
+    imageSrc: string;
+    href: string;
+    title: string;
+    body: string;
+    showInitialText: boolean;
+    textHideable: boolean;
+    imageOnRight: boolean;
+    isTimelineCard: boolean;
+}
+
+export function Card({ imageSrc, href, title, body, showInitialText, textHideable, imageOnRight, isTimelineCard }: CardProps) {
+
+    const [textVisible, setTextVisible] = useState(showInitialText);
+
+    function changeToggleText(){
+        if(textHideable){
+            setTextVisible(!textVisible);
+        }  
+    }
+
+    const cardContent = (
+        <div className={`card-link card-image-on-${imageOnRight ? "right" : "left"}`}>
+            <img className="card card-image" src={imageSrc} onClick={changeToggleText} />
+            { textVisible && 
+                <div className="card-text-block">
+                    <h1 className="card card-title">{title}</h1>
+                    <p className="card card-body">{body}</p>
+                </div> }
+        </div>
+    );
+
+    if (href !== "") {
+        return (
+            <div className={`card card-object card-is-in-timeline-${isTimelineCard}`} >
+                <Link to={href}>
+                    {cardContent}
+                </Link>
+            </div>
+        )
+    }
+
+    return (
+        <div className={`card card-object card-is-in-timeline-${isTimelineCard}`}>
+            <div className={`card-link card-image-on-${imageOnRight ? "right" : "left"}`}>
```

Comment:
The link is gone?

Reply:
the link is only requied on the return for when a link is provided, otherwise a link is not needed and only a div is returned. It renders correctly at the moment.